### DISCUSSION TITLE
On the lighter side. Oqtane on a Pi.

### DISCUSSION THREAD

**Discussion by louyo** [oqtane/oqtane.framework#737] [General]
![oqtanePi](https://user-images.githubusercontent.com/6570793/92672623-10b6bd00-f2e7-11ea-9171-cb232408ef88.png)

For those with too much time on their hands

**Comment by wcrisafu** [comment]
Hi Louyo,

Could you share the installation steps? especialy details around the SQL DB.

**Comment by louyo** [comment]
Well, I can't believe anyone else is a nutty as me:
Ok, as best I remember. I did this just to see what worst case performance would be. It is quite good. 
I did not try to put the DB on the Pi, I had to run the 32 bit version of Linux on the Pi. I had SQL Server running on a Windws server. I don't think there is a 32 bit version and I am not sure about arm. This is also, basically, what I did to install on my Ubuntu virtual machine. Although it is ]64 bit.

I did (briefly) try the 64 bit Ubuntu server  image on the Pi, but could not get Oqtane to run. Maybe my shortcoming. Since I was mostly experimenting, here is what I did.

Installed the .net SDK on the Pi. There is a 32 bit Arm version. Followed the Linux directions at Microsoft.
ran dotnet --version to test.

Copied over the Oqtane-master directories into  ~/ftp (which  I had created).

added: .UseUrls("http://*:5000") to the Program.cs file in the Oqtane.Server folder

In appsettings.json, put:
"DefaultConnection": "Data Source=10.0.2.67;Initial Catalog=oqtane;User ID=sa;Password=<mypassword>" for the connection string.
where my SQL Server is on 10.0.2.67 (with remote access enabled) and using my real sa password.

Then, from /home/pi/ftp/oqtane.framework-master/ I ran:
dotnet publish -c Release -r linux-arm -o ~/ftp/oqtane

Then: from the ftp/oqtane directory, I ran chmod +x Oqtane.Server
and finally: ./Oqtane.Server
If the  Oqtane and Pi gods are friendly, you will see:

Hosting environment: Production
Content root path: /home/pi/ftp/oqtane
Now listening on: http://[::]:5000
Application started. Press Ctrl+C to shut down.

Happy  Oqtaning.

**Reply by wcrisafu** [reply]
Right now I am getting up to speed on my PC but my goal is to have Oqtane & my module working on a PI4. I got Core to run on my PI4 but SQL server could not run on ARM yet. Hopefully by the time I finish my module we will have either a new SQL version or Oqtane will work on SQLite.

**Reply by thabaum** [reply]
Cool!

**Comment by thabaum** [comment]
I was looking to do some project like this, very cool!
